### ISSUE TITLE
durability in a cluster

### ISSUE THREAD

**Issue by deviantmk** [rethinkdb/rethinkdb#5510]
Hi
while using **soft durability** and full cluster replication (n/n), can data loss occur if a machine fails?

i skimmed through the docs

> Consider using soft durability mode. In soft durability mode RethinkDB will acknowledge the write immediately after receiving it, but before the write has been committed to disk. The server will use main memory to absorb the write, and will flush new data to disk in the background.

so will the write be immediately acknowledge on every cluster node, or it will propagate to the rest of the nodes only after its written to disk on the initial one

**Comment by danielmewes** [comment]
If you have a cluster of at least two servers, and a single server fails, you should not get any data loss even with soft durability.

The reason is that a write will only be acknowledged once it has been applied on a majority of replicas.

Soft durability changes the meaning of "has been applied" in this, but the write still has to go to a majority first before the acknowledgement. Basically the write will still be applied in memory on a majority before it's acked. Only the writing to disk might be delayed.

The documentation is a bit misleading in this detail. I'll open a new issue on our docs issue tracker to address that. https://github.com/rethinkdb/docs/issues

**Comment by danielmewes** [comment]
>  so will the write be immediately acknowledge on every cluster node, or it will propagate to the rest of the nodes only after its written to disk on the initial one

It will be send to all replicas at the same time, regardless of when it gets written to disk.

**Comment by danielmewes** [comment]
https://github.com/rethinkdb/docs/issues/1048
